client/client.go

```diff
@@ -316,11 +317,76 @@ func makeQueryString(data interface{}) (string, error) {
 	return "", fmt.Errorf("unsupported query string data: %#v", data)
 }
 
+// Remove all custom request serializer logic once APP-1331 is rolled out.
+type serializer func(any) ([]byte, error)
+
+func serializeUnityCatalogDeleteRequest(data any) ([]byte, error) {
```

Comment:
i think it's better to remove `UnityCatalog` from the name of the function. we may have some of the non-compliant GET requests as well.